### PR TITLE
Pass context with HTTP request

### PR DESCRIPTION
**Pull request by NathanBaulch**

### CODE REVIEW

**Comment by NathanBaulch** [comment]
Unit test added.

File: notificationhub.go [hunk 1]
```diff
@@ -114,7 +114,7 @@ func (h *NotificationHub) exec(ctx context.Context, method string, url *url.URL,
 	if err != nil {
 		return nil, nil, err
 	}
-	req.WithContext(ctx)
+	req = req.WithContext(ctx)
```
Comment (kurtsson) [inline notificationhub.go:117]: I would rather use `http.NewRequestWithContext(`on line 113

**Review by kurtsson — CHANGES_REQUESTED** [review]
Good change, but not so pretty solution

File: notificationhub.go (hunk 1 above)
Comment (NathanBaulch) [reply notificationhub.go:117]: Looks like this function was introduced in [Go 1.13](https://golang.org/doc/go1.13#net/http).
Travis tests against Go 1.10, 1.11 and 1.12 which is why it's now failing.

**Comment by NathanBaulch** [comment]
I can squash the "Backward compat idea" commit if you like that approach, or revert to the previous `WithContext` func if you want to keep it simple.

File: notificationhub.go (hunk 1 above)
Comment (kurtsson) [reply notificationhub.go:117]: Good catch! Thank you

**Comment by kurtsson** [comment]
> I can squash the "Backward compat idea" commit if you like that approach, or revert to the previous `WithContext` func if you want to keep it simple.

I rather keep it backwards compatible, it doesn't add any overhead anyway exept a line of code. So please revert again, and sorry for the fuzz

File: notificationhub_go1_13.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+package notificationhubs
+
+import (
+	"context"
+	"io"
+	"net/http"
+)
+
+func newHTTPRequest(ctx context.Context, method, url string, body io.Reader) (*http.Request, error) {
+	return http.NewRequestWithContext(ctx, method, url, body)
```
Comment (kurtsson) [inline notificationhub_go1_13.go]: It's better to keep it backwards compatible

**Review by kurtsson — CHANGES_REQUESTED** [review]

**Review by kurtsson — APPROVED** [review]
👍

**Pull request merged**